os/drivers/spi/spi_uio.c

```diff
@@ -0,0 +1,233 @@
+ * Name: spidrvr_close
+ ****************************************************************************/
+
+static int spi_uio_close(FAR struct file *filep)
+{
+	SPI_ENTRY;
+	FAR struct inode *inode;
+	FAR struct spi_dev_s *dev;
+
+	/* Get our private data structure */
+
+	DEBUGASSERT(filep != NULL && filep->f_inode != NULL);
+	inode = filep->f_inode;
+
+	dev = (FAR struct spi_dev_s *)inode->i_private;
+	DEBUGASSERT(dev);
+
+	return 0;
+}
+
+
+/****************************************************************************
+ * Name: spi_uioread
+ *
+ * Description:
+ *   Read SPI Data. This function is automatically called when you're doing
+ *   read() API on userspace.
+ *
+ ****************************************************************************/
+
+static ssize_t spi_uioread(FAR struct file *filep, FAR char *buffer, size_t buflen)
+{
+	SPI_ENTRY;
+	FAR struct inode *inode = filep->f_inode;
+	FAR struct spi_dev_s *dev = inode->i_private;
+
+	SPI_LOCK(dev, true);
+	SPI_SELECT(dev, dev->config.cs, true);
+	SPI_RECVBLOCK(dev, buffer, buflen); // return type is void
+	SPI_SELECT(dev, dev->config.cs, false);
+	SPI_LOCK(dev, false);
+
+	return 0;
+}
+
+/****************************************************************************
+ * Name: spi_uiowrite
+ *
+ * Description:
+ *   Write SPI Data. This function is automatically called when you're doing
+ *   write() API on userspace.
+ *
+ ****************************************************************************/
+
+static ssize_t spi_uiowrite(FAR struct file *filep, FAR const char *buffer, size_t buflen)
+{
+	SPI_ENTRY;
+	FAR struct inode *inode = filep->f_inode;
+	FAR struct spi_dev_s *dev = inode->i_private;
+
+	SPI_LOCK(dev, true);
+	SPI_SELECT(dev, dev->config.cs, true);
+	SPI_SNDBLOCK(dev, buffer, buflen); // return type is void
+	SPI_SELECT(dev, dev->config.cs, false);
+	SPI_LOCK(dev, false);
+
+	return 0;
+}
+
+/****************************************************************************
+ * Name: spi_uioctrl
+ *
+ * Description:
+ *   Control SPI Data. This function is automatically called when you're doing
+ *   ctrl() API on userspace.
+ *
+ ****************************************************************************/
+/* struct _spi_config { */
+/* 	int chip_select; */
+/* 	handle->bpw = config->bits_per_word; */
+/* 	handle->freq = config->frequency; */
+/* 	handle->cs = config->chip_select; */
+/* 	handle->mode = config->mode; */
+/* }; */
+
+/* typedef struct _spi_io_config { */
+/* 	int chip_select; */
+/* 	unsigned int bpw; */
+/* 	int freq; // clock speed */
+/* 	int cs; */
+/* 	spi_mode_e mode; */
+/* } spi_io_config; */
+
+/* struct spi_io_config { */
+/* 	unsigned int bpw; */
+/* 	int freq; // clock speed */
+/* 	int cs; */
+/* 	enum spi_mode_e mode; */
+/* }; */
+
+static int spi_uioctrl(FAR struct file *filep, int cmd, unsigned long arg)
+{
+	SPI_ENTRY;
+
+	FAR struct inode *inode = filep->f_inode;
+	FAR struct spi_dev_s *dev = inode->i_private;
+
+	if (cmd == SPI_IO_CONFIG) {
```

Comment:
Please apply ioctl's command style. 
In this case, SPIIOC_CONFIGURE will be better.

Reply:
Thanks I'll fix it